fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/MockedFDBDirectory.java

```diff
@@ -50,10 +52,25 @@ public enum Methods { GET_INCREMENT,
         LIST_ALL,
         GET_FILE_REFERENCE_CACHE_ASYNC,
         DELETE_FILE_INTERNAL,
-        GET_PRIMARY_KEY_SEGMENT_INDEX
+        GET_PRIMARY_KEY_SEGMENT_INDEX,
+        GET_ALL_FIELDS_INFO_STREAM
     }
 
-    private EnumMap<Methods, FailureDescription> failureDescriptions = new EnumMap<>(Methods.class);
+    /**
+     * The scope used to add failures.
+     * LOCAL only impacts the mocked directory on which the call is made
+     * GLOBAL impacts all mocked directories loaded by the class loader
+     *
+     * NOTE: Since GLOBAL failures are placed in a static variable, setting this in a test requires it being cleaned up
```

Comment:
Note: this won't play well with concurrent tests. Without removing the global scope you could fix the concurrency issue specifically by adding a `ResourceLock` for anything that interacts with `MockedFDBDirectory`: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parallel-execution-synchronization

Reply:
GLOBAL state was removed as state is now kept at the IndexMaintainerFactory